### PR TITLE
refactor(skore): Summary concatenation helper

### PR DESCRIPTION
**Pull request by cakedev0**
Context: I started this refactor while working on the "export results" feature.

#### Change description

Changes on `MetricsSummaryDisplay`:
- Added a `_concatenate` static method, to centralize concatenation logic.
- Change input from a dataframe to rows (will be convenient so I can reuse the summary logic in the "export results" feature). Also, it simplifies concatenation logic.
- Kept a `.data` property to avoid having to edit many places/tests. But maybe we can remove it later (let me know what you think).

+ Adapted code in metric accessors to those changes (simplifies it).

### CODE REVIEW

File: skore/src/skore/_sklearn/_plot/metrics/metrics_summary_display.py [hunk 1]
```diff
@@ -1,11 +1,12 @@
-from typing import Literal, cast
+from typing import Any, Literal, cast
 
 import pandas as pd
 from matplotlib.figure import Figure
 
 from skore._sklearn._plot.base import DisplayMixin
 from skore._sklearn.types import Aggregate, ReportType
 from skore._utils._index import flatten_multi_index
+from skore._utils._metric_rows import rows_to_dataframe
```
Comment (cakedev0) [inline skore/src/skore/_sklearn/_plot/metrics/metrics_summary_display.py:9]: Note: I think we could move functions from `_metric_rows` to here, that would make sense. But it will grow artificially the diff, so I don't do it in this PR.

Comment (auguste-probabl) [reply skore/src/skore/_sklearn/_plot/metrics/metrics_summary_display.py:9]: sure!

File: skore/src/skore/_sklearn/_plot/metrics/metrics_summary_display.py [hunk 2]
```diff
@@ -26,12 +39,29 @@ class MetricsSummaryDisplay(DisplayMixin):
 
     def __init__(
         self,
-        data: pd.DataFrame,
+        rows: list[dict],
```
Comment (auguste-probabl) [inline skore/src/skore/_sklearn/_plot/metrics/metrics_summary_display.py:42]: I really like `TypedDict` to document structured dictionaries, could you use that and move the Parameters description there?

**Review by auguste-probabl — APPROVED** [review]
A small suggestion on typing/documentation but this looks great!

File: skore/src/skore/_sklearn/_plot/metrics/metrics_summary_display.py (hunk 2 above)
Comment (cakedev0) [reply skore/src/skore/_sklearn/_plot/metrics/metrics_summary_display.py:42]: I'll merge it first, then I'll do that in a small follow-up PR where I also move functions from `_metric_rows.py` in `metrics_summary_display.py`.
